src/app/app.states.ts

```diff
@@ -0,0 +1,16 @@
+import { AppComponent } from './app.component';
+import { HomepageComponent } from './modules/homepage/views/homepage/homepage.component';
+import { FooViewComponent } from './modules/foo/views/foo-view/foo-view.component';
+import { BarViewComponent } from './modules/bar/views/bar-view/bar-view.component';
+import { NotFoundComponent } from './modules/not-found/not-found.component';
+
+export const APP_STATES = [
+    { name: 'app', redirectTo: 'homepage', component: AppComponent },
+    { parent: 'app', name: 'homepage', url: '/home', component: HomepageComponent },
+    { parent: 'app', name: 'foo.**', url: '/foo', loadChildren: './modules/foo/foo.module#FooModule' },
```

Comment:
Did you test the modules' lazy load feature? Does it work with the UI-router version?

Reply:
Yes, lazy loading works fine, but I don't think preloading is supported in the current version of the ui-router.